engine/src/main/java/org/terasology/rendering/dag/nodes/OutputToScreenNode.java

```diff
@@ -34,29 +36,85 @@
+    private WorldRenderer worldRenderer;
+
+    private FBO lastUpdatedGBuffer;
+    private FBO staleGBuffer;
+
+    private StateChange bindFbo;
+
+    public OutputToScreenNode(Context context) {
         super(context);
 
+        displayResolutionDependentFBOs = context.get(DisplayResolutionDependentFBOs.class);
+        worldRenderer = context.get(WorldRenderer.class);
+
         WorldRenderer worldRenderer = context.get(WorldRenderer.class);
         requiresCondition(() -> worldRenderer.getCurrentRenderStage() == MONO || worldRenderer.getCurrentRenderStage() == LEFT_EYE);
 
         addDesiredStateChange(new EnableMaterial(DEFAULT_TEXTURED_MATERIAL_URN));
 
-        addDesiredStateChange(new SetInputTextureFromFbo(0, FINAL_BUFFER, ColorTexture,
-                context.get(DisplayResolutionDependentFBOs.class), DEFAULT_TEXTURED_MATERIAL_URN, "texture"));
+        bindFbo = new SetInputTextureFromFbo(0, FINAL_BUFFER, ColorTexture, displayResolutionDependentFBOs, DEFAULT_TEXTURED_MATERIAL_URN, "texture");
+        addDesiredStateChange(bindFbo);
+
+        SwappableFBO gBufferPair = displayResolutionDependentFBOs.getGBufferPair();
+        lastUpdatedGBuffer = gBufferPair.getLastUpdatedFbo();
+        staleGBuffer = gBufferPair.getStaleFbo();
     }
 
     @Override
     public void process() {
-        PerformanceMonitor.startActivity("rendering/copyImageToScreen");
+        PerformanceMonitor.startActivity("rendering/outputToScreen");
         // The way things are set-up right now, we can have FBOs that are not the same size as the display (if scale != 100%).
         // However, when drawing the final image to the screen, we always want the viewport to match the size of display,
         // and not that of some FBO. Hence, we are manually setting the viewport via glViewport over here.
         glViewport(0, 0, Display.getWidth(), Display.getHeight());
         renderFullscreenQuad();
         PerformanceMonitor.endActivity();
     }
+
+    @Override
+    public void handleCommand(String command, String... arguments) {
+        switch (command) {
+            case "setFbo":
+                if (arguments.length != 1) {
+                    throw new RuntimeException("Invalid number of arguments; expected 1, received " + arguments.length + "!");
+                }
+
+                FBO fbo;
+                switch (arguments[0]) {
+                    case "engine:fbo.gBuffer":
+                    case "engine:fbo.lastUpdatedGBuffer":
+                        fbo = lastUpdatedGBuffer;
+                        break;
+                    case "engine:fbo.staleGBuffer":
+                        fbo = staleGBuffer;
+                        break;
+                    default:
+                        fbo = displayResolutionDependentFBOs.get(new SimpleUri(arguments[0]));
+
+                        if (fbo == null) {
+                            throw new RuntimeException("Invalid fbo uri: " + arguments[0] + "!");
```

Comment:
Let's be a bit more specific here:

`("No FBO is associated with URI '" + arguments[0] + "'")`